src/PlaybackManager.vala

```diff
@@ -22,6 +23,14 @@ public class Music.PlaybackManager : Object {
     private uint progress_timer = 0;
     private Settings settings;
 
+    private enum Direction {
+        NONE,
+        NEXT,
+        PREVIOUS
+    }
+
+    private Direction direction = Direction.NONE;
```

Comment:
This seems kind of fragile to me. Do we still need this since we're no longer adding the files into the queue? Is it safe to just assume we also go `next ()`?

Reply:
I think this might still be needed, because invalid tracks can still get added to the Q, I mean the `GLib.ContentType.guess ()` appears to just be looking at the file extension, I can still get a text file renamed as mp3 added to the Q. Which implies if there are corrupted audio tracks they would require this handling.

Also, people could click the previous button too to go back to earlier tracks, right?

Reply:
I'll leave things here as is, or I could remove the `NONE` enum and initialise the var to `NEXT` instead?